### PR TITLE
Update data fix

### PR DESCRIPTION
**Pull request by mrocklin**

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @pitrou

File: distributed/scheduler.py [hunk 1]
```diff
@@ -1622,11 +1630,13 @@ def update_data(self, stream=None, who_has=None, nbytes=None, client=None):
                 if key not in self.dependencies:
                     self.dependencies[key] = set()
                 self.task_state[key] = 'memory'
-                self.who_has[key] = set(workers)
+                if key not in self.who_has:
+                    self.who_has[key] = set()
```
Comment (mrocklin) [inline distributed/scheduler.py]: This was the actual error.

File: distributed/scheduler.py [hunk 2]
```diff
@@ -647,9 +647,9 @@ def stimulus_task_finished(self, key=None, worker=None, **kwargs):
 
         if self.task_state[key] == 'memory':
             self.who_has[key].add(worker)
+            self.has_what[worker].add(key)
             if key not in self.has_what[worker]:
```
Comment (pitrou) [inline distributed/scheduler.py]: This line is now pointless?

Comment (mrocklin) [inline distributed/scheduler.py]: Agreed.  Thanks for catching.  Fixed.

**Pull request merged**